homeassistant/components/swidget/manifest.json

```diff
@@ -0,0 +1,26 @@
+{
+  "domain": "swidget",
+  "name": "Swidget",
+  "codeowners": ["@michaelkkehoe"],
+  "config_flow": true,
+  "dependencies": ["network"],
+  "dhcp": [
+    {
+      "registered_devices": true
+    },
+    {
+      "hostname": "swidget*",
+      "macaddress": "24A160*"
+    }
+  ],
+  "documentation": "https://www.home-assistant.io/integrations/swidget",
+  "integration_type": "device",
+  "iot_class": "local_push",
```

Comment:
where is the push happening?

Reply:
The Swidget API client creates a websocket to the device. The device will automatically push change notifications back to the Swidget API client over websocket (so you don't have to poll it for changes).